exporter/elasticsearchexporter/config.go

```diff
@@ -159,6 +159,9 @@ type MappingsSettings struct {
 	// Try to find and remove duplicate fields
 	Dedup bool `mapstructure:"dedup"`
 
+	// Deprecated: [v0.104.0] dedotting will always be applied for ECS mode
```

Comment:
nit: I think this deprecation won't be visible to users' configs from yaml. You may want to log a warning if this is set.

Reply:
Yep, good call, thanks. There's a limited amount of opportunity to warn since the default behaviour will change, but I've added in a warning for what makes sense.